pythonforandroid/recipe.py

```diff
@@ -213,24 +213,26 @@ def report_hook(index, blksize, size):
                 break
             return target
         elif parsed_url.scheme in ('git', 'git+file', 'git+ssh', 'git+http', 'git+https'):
-            if isdir(target):
-                with current_directory(target):
-                    shprint(sh.git, 'fetch', '--tags', '--recurse-submodules')
-                    if self.version:
-                        shprint(sh.git, 'checkout', self.version)
-                    branch = sh.git('branch', '--show-current')
-                    if branch:
-                        shprint(sh.git, 'pull')
-                        shprint(sh.git, 'pull', '--recurse-submodules')
-                    shprint(sh.git, 'submodule', 'update', '--recursive')
-            else:
+            if not isdir(target):
                 if url.startswith('git+'):
                     url = url[4:]
-                shprint(sh.git, 'clone', '--recursive', url, target)
+                # if 'version' is specified, do a shallow clone
                 if self.version:
+                    shprint(sh.mkdir, '-p', target)
                     with current_directory(target):
-                        shprint(sh.git, 'checkout', self.version)
-                        shprint(sh.git, 'submodule', 'update', '--recursive')
+                        shprint(sh.git, 'init')
+                        shprint(sh.git, 'remote', 'add', 'origin', url)
```

Comment:
have we thought about directly cloning that branch from here and skipping the other code path?
Something like:
```sh
shprint(sh.git, 'clone', '--branch', self.version, '--single-branch', url)
```
I haven't tested so I don't know if the rest of the code apply something we would need still

Reply:
Which code path do you mean?
Do you mean you would do this instead of
```python
shprint(sh.git, 'fetch', '--depth', '1', 'origin', self.version)
shprint(sh.git, 'checkout', self.version)
```

but note that `version` is not necessarily a branch -- I for example always set it to a commit hash, for reproducibility. In that case your command errors.

Also, btw,  I could simplify this logic if we raised on `version` not being set. I am not sure in what scenario it makes sense to set a git repo but not even specify a branch.

The combinatorial matrix of possibilities is surprisingly large here btw,
`{isdir(target)}`x`{version in (not_set, branch, commit)}` gives 6 options, and that's not even counting whether there are submodules.

Reply:
Makes sense, thanks :+1: